src/user/user.service.ts

```diff
@@ -30,19 +38,20 @@ export class UserService {
     @InjectRepository(UserRepository)
     private userRepository: UserRepository,
     private readonly partnerAccessService: PartnerAccessService,
+    private readonly partnerService: PartnerService,
     private readonly authService: AuthService,
   ) {}
 
   public async createUser(createUserDto: CreateUserDto): Promise<GetUserDto> {
-    const {
-      name,
-      email,
-      partnerAccessCode,
-      contactPermission,
-      signUpLanguage,
-      partnerId,
-      password,
-    } = createUserDto;
+    const { email, partnerAccessCode, partnerId, password } = createUserDto;
+
+    const signUpType =
```

Comment:
Optional: This is a style note but I'm wondering if an `if` statement here checking on partnerId first to separate between public and partnered users would help readability. 
`if(partnerId){
 // two flavours
 } else {
// public
}
`

Reply:
Yeah theres a mutability vs readability conundrum!

Reply:
I've switched them round so perhaps it might read easier